src/main/java/com/couchbase/lite/internal/Body.java

```diff
@@ -126,8 +124,9 @@ public String getJSONString() {
     public Map<String, Object> getProperties() {
         Object object = getObject();
         if(object instanceof Map) {
-            Map<String, Object> map = (Map<String, Object>) object;
-            return Collections.unmodifiableMap(map);
```

Comment:
@pasin I believe the only reason was trying to emulate the same behavior as on couchbase-lite-ios